smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServerGenerator.java

```diff
@@ -46,114 +46,202 @@ static void generateServiceHandler(SymbolProvider symbolProvider,
             writer.write("this.service = service;");
             writer.write("this.mux = mux;");
             writer.write("this.serializerFactory = serializerFactory;");
             writer.write("this.serializeFrameworkException = serializeFrameworkException;");
             writer.closeBlock("}");
-            writer.openBlock("async handle(request: HttpRequest): Promise<HttpResponse> {", "}", () -> {
+            writer.openBlock("async handle(request: __HttpRequest): Promise<__HttpResponse> {", "}", () -> {
                 writer.write("const target = this.mux.match(request);");
                 writer.openBlock("if (target === undefined) {", "}", () -> {
-                    writer.write("return serializeFrameworkException(new UnknownOperationException(), "
-                            + "this.serdeContextBase);");
+                    writer.write("return this.serializeFrameworkException(new __UnknownOperationException(), "
+                            + "serdeContextBase);");
                 });
                 writer.openBlock("switch (target.operation) {", "}", () -> {
                     for (OperationShape operation : operations) {
-                        generateHandlerCase(writer, operation, symbolProvider.toSymbol(operation));
+                        String opName = operation.getId().getName();
+                        writer.openBlock("case $S : {", "}", opName, () -> {
+                            writer.write("return handle(request, this.serializerFactory($S), this.service.$L, "
+                                    + "this.serializeFrameworkException);",
+                                    opName, symbolProvider.toSymbol(operation).getName());
+                        });
                     }
                 });
             });
         });
     }
 
-    private static void generateHandlerCase(TypeScriptWriter writer,
-                                            Shape operationShape,
-                                            Symbol operationSymbol) {
-        String opName = operationShape.getId().getName();
-        writer.openBlock("case $S : {", "}", opName, () -> {
-            writer.write("let serializer = this.serializerFactory($S);", opName);
-            writer.write("let input;");
-            writer.openBlock("try {", "} catch (error: unknown) {", () -> {
-                writer.openBlock("input = await serializer.deserialize(request, {", "});", () -> {
-                    writer.write("endpoint: () => Promise.resolve(request), ...this.serdeContextBase");
-                });
+    static void generateOperationHandler(SymbolProvider symbolProvider,
+                                                 Shape serviceShape,
+                                                 OperationShape operation,
+                                                 TypeScriptWriter writer) {
+        addCommonHandlerImports(writer);
+
+        writeSerdeContextBase(writer);
+        writeHandleFunction(writer);
```

Comment:
Can't these be somewhere shared?

Reply:
Do you mean written to one TS file that's shared between the two? Seemed more complicated than it was worth to me at the time. Besides, the duplicates should always be tree-shaken out, right?